### PR TITLE
[receiver/github] update metrics and attributes to match latest semantics and change scraper key name

### PR DESCRIPTION
**Pull request by adrielp**
#### Description

Breaking: Updates various different metrics and attributes to match the latest semantic conventions (1.28+). Also updates the scraper key name.

Most of the conventions are in 1.28 while a few attributes are merged in and will be released in 1.29.

* Change the `github` scraper key to `scraper`
* Add `vcs.repository.url.full` attribute
* Change attribute `repository.name` to `vcs.repository.name`
* Change attribute `ref.name` to `vcs.ref.head.name`
* Change attribute `ref.type` to `vcs.ref.head.type`
* Change attribute `change.state` to `vcs.change.state`
* Add attribute `vcs.revision_delta.direction` with `ahead` and `behind` values
* Change metric `vcs.repository.ref.revisions_ahead` to `vcs.ref.revisions_delta` with `vcs.revision_delta.direction=ahead`
* Change metric `vcs.repository.ref.revisions_behind` to `vcs.ref.revisions_delta` with `vcs.revision_delta.direction=behind`
* Change metric `vcs.repository.ref.count` to `vcs.ref.count`
* Change metric `vcs.repository.ref.time` to `vcs.ref.time`
* Add attribute `vcs.line_change.type` with `added` and `removed` values
* Change metric `vcs.repository.ref.lines_added` to `vcs.ref.lines_delta` with `vcs.line_change.type=added`
* Change metric `vcs.repository.ref.lines_removed` to `vcs.ref.lines_delta` with `vcs.line_change.type=removed`
* Change metric `vcs.repository.contributor.count` to `vcs.contributor.count`
* Change metric `vcs.repository.change.time_open` to `vcs.change.duration` with `vcs.change.state=open`
* Change metric `vcs.repository.change.time_to_approval` to `vcs.change.time_to_approval`
* Change metric `vcs.repository.change.time_to_merge` to `vcs.change.time_to_merge`
* Change metric `vcs.repository.change.count` to `vcs.change.count`

#### Testing

In addition to the normal testing of the code, I additionally built the receiver into a collector to observe runtime behavior.

#### Documentation

Update generated docs and readme with scraper change.

### CODE REVIEW

**Comment by adrielp** [comment]
Upon investigation, the `githubgen` check within the code owners workflow failure is unrelated to my changes and is broken across the board. Active related work there is #36023 #36285

File: receiver/githubreceiver/README.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ receivers:
         initial_delay: 1s
         collection_interval: 60s
         scrapers:
-            github:
+            scraper:
```
Comment (andrzej-stencel) [inline receiver/githubreceiver/README.md:58]: Why change the name of this scraper? Now that this receiver is named "GitHub receiver" and not "Git Provider receiver", I suppose there will be only one scraper in this receiver, is this correct? If yes, the configuration should not have `scrapers::github` or `scrapers:scraper` at all, but rather the `metrics` configuration option should be a top-level option in the receiver. Does it make sense?

Comment (andrzej-stencel) [reply receiver/githubreceiver/README.md:58]: Oh, interesting, so you see a use case in defining more than one scraper in a single receiver. I suppose this is OK then. Perhaps we can add more examples in the docs to make it clearer for useres (as a separate contribution). Thanks for explaining this!

**Review by andrzej-stencel — APPROVED** [review]

**Pull request merged**